src/test_suite/test_suite.py

```diff
@@ -485,5 +488,30 @@ def decode_protobuf(
     print(f"{sum(write_results)} files successfully written")
 
 
+def set_ld_preload_asan():
+    # Run ldconfig -p and capture output
+    import subprocess
+
+    ldconfig_output = subprocess.check_output(["ldconfig", "-p"], text=True)
+
+    # Filter lines containing "asan"
+    asan_libs = [line for line in ldconfig_output.split("\n") if "asan" in line]
+
+    # Extract the first library path if available
+    if asan_libs:
+        first_asan_lib = asan_libs[0].split()[-1]
+    else:
+        print("No ASAN library found.")
+        return
```

Comment:
Can you confirm loading just the ASAN library is enough to let us run FD with extras (e.g. ubsan, fuzz)? Otherwise we may need to load in other preloads as well

Reply:
I was planning on adding the others as we encounter issues, the ASAN library is the only one I remember causing this issue.